src/realm/sync/noinst/client_reset.cpp

```diff
@@ -499,8 +509,995 @@ void EmbeddedObjectConverter::process_pending()
+                bool did_translate = resolve(it.first, [&](LstBase& remote_list, LstBase& local_list) {
+                    ConstTableRef local_table = local_list.get_table();
+                    ConstTableRef remote_table = remote_list.get_table();
+                    ColKey local_col_key = local_list.get_col_key();
+                    ColKey remote_col_key = remote_list.get_col_key();
+                    Obj local_obj = local_list.get_obj();
+                    Obj remote_obj = remote_list.get_obj();
+                    InterRealmValueConverter value_converter(local_table, local_col_key, remote_table, remote_col_key,
+                                                             embedded_object_tracker);
+                    logger.debug("Recovery overwrites list for '%1' size: %2 -> %3", path_str, remote_list.size(),
+                                 local_list.size());
+                    copy_list(local_obj, remote_obj, value_converter, nullptr);
+                    embedded_object_tracker.process_pending();
+                });
+                if (!did_translate) {
+                    // object no longer exists in the local state, ignore and continue
+                    logger.warn("Discarding a list recovery made to an object which could not be resolved. "
+                                "remote_path='%3'",
+                                path_str);
+                }
+            }
+        }
+        embedded_object_tracker.process_pending();
+    }
+
+    bool resolve_path(ListPath& path, Obj remote_obj, Obj local_obj,
+                      util::UniqueFunction<void(LstBase&, LstBase&)> callback)
+    {
+        for (auto it = path.m_path.begin(); it != path.m_path.end();) {
+            if (!remote_obj || !local_obj) {
+                return false;
+            }
+            REALM_ASSERT(it->type == ListPath::Element::Type::ColumnKey);
+            ColKey col = it->col_key;
+            REALM_ASSERT(col);
+            if (col.is_list()) {
+                std::unique_ptr<LstBase> remote_list = get_list_from_path(remote_obj, col);
+                ColKey local_col =
+                    local_obj.get_table()->get_column_key(remote_obj.get_table()->get_column_name(col));
+                REALM_ASSERT(local_col);
+                std::unique_ptr<LstBase> local_list = get_list_from_path(local_obj, local_col);
+                ++it;
+                if (it == path.m_path.end()) {
+                    callback(*remote_list, *local_list);
+                    return true;
+                }
+                else {
+                    REALM_ASSERT(it->type == ListPath::Element::Type::ListIndex);
+                    REALM_ASSERT(it != path.m_path.end());
+                    size_t stable_index_id = it->index;
+                    REALM_ASSERT(stable_index_id != realm::npos);
+                    // This code path could be implemented, but because it is currently not possible to
```

Comment:
Why is it not possible to exercise in tests? If a customer could ever hit this we should throw an exception here to be handled rather than crashing the app. If this is impossible, do we need the checks and stable_index_id variable above?

Reply:
It should not be possible to get to this code path at all. I could take out the extra asserts, but I also kinda feel like they might help if the code changes in such a way that this becomes possible.